src/azure-cli/azure/cli/command_modules/synapse/commands.py

```diff
@@ -137,5 +169,84 @@ def get_custom_sdk(custom_module, client_factory):
         g.custom_command('list', 'list_role_definitions')
         g.custom_show_command('show', 'get_role_definition')
 
+    # Data Plane Commands --Artifacts Linked service operations
+    with self.command_group('synapse linked-service', synapse_linked_service_sdk,
+                            custom_command_type=get_custom_sdk('artifacts', None)) as g:
+        g.custom_command('create', 'create_or_update_linked_service', supports_no_wait=True)
+        g.custom_command('set', 'create_or_update_linked_service', supports_no_wait=True)
+        g.custom_command('list', 'list_linked_service')
+        g.custom_show_command('show', 'get_linked_service')
+        g.custom_command('delete', 'delete_linked_service', confirmation=True, supports_no_wait=True)
+
+    # Data Plane Commands --Artifacts dataset operations
+    with self.command_group('synapse dataset', synapse_dataset_sdk,
+                            custom_command_type=get_custom_sdk('artifacts', None)) as g:
+        g.custom_command('create', 'create_or_update_dataset', supports_no_wait=True)
+        g.custom_command('set', 'create_or_update_dataset', supports_no_wait=True)
+        g.custom_command('list', 'list_datasets')
+        g.custom_show_command('show', 'get_dataset')
+        g.custom_command('delete', 'delete_dataset', confirmation=True, supports_no_wait=True)
+
+    # Data Plane Commands --Artifacts pipeline operations
+    with self.command_group('synapse pipeline', synapse_pipeline_sdk,
+                            custom_command_type=get_custom_sdk('artifacts', None)) as g:
+        g.custom_command('create', 'create_or_update_pipeline', supports_no_wait=True)
+        g.custom_command('set', 'create_or_update_pipeline', supports_no_wait=True)
+        g.custom_command('list', 'list_pipelines')
+        g.custom_show_command('show', 'get_pipeline')
+        g.custom_command('delete', 'delete_pipeline', confirmation=True, supports_no_wait=True)
+        g.custom_command('create-run', 'create_pipeline_run')
+
+    # Data Plane Commands --Artifacts pipeline run operations
+    with self.command_group('synapse pipeline-run', synapse_pipeline_run_sdk,
+                            custom_command_type=get_custom_sdk('artifacts', None)) as g:
+        g.custom_command('query-by-workspace', 'query_pipeline_runs_by_workspace')
```

Comment:
For simplicity, using "query" as command is enough?

Reply:
It keep align with ADF. In ADF, it is "datafactory pipeline-run query-by-factory".